### DISCUSSION TITLE
Static binaries for dendrite?

### DISCUSSION THREAD

**Discussion by lammel** [matrix-org/dendrite#2432] [Help & Support]
Is there a chance to build static binaries for dendrite?
I'm pretty used to just building a static binary with go and copy it to a server. 
As some parts are depending on C libraries (e.g. sqlite) requiring CGO for buiding. So all produced binaries are dynamically linked against libc which requires the same libc to be available on the target system.

Did anyone attempt static builds already?

**Comment by neilalexander** [comment]
You could try `go build -ldflags="-extldflags=-static"`, but you might still run into problems with different `glibc` versions.

You could also try combining the above with installing musl, telling Go to use the `musl-gcc` compiler by setting the `CC` variable when running `go build`.
